features/steps/steps.py

```diff
@@ -413,3 +485,64 @@ def step_impl(context, entity, other_entities):
                 errors.append(instance_structure_error(inst, [i for i in nested_entities if i.is_a() in differences], 'nested by'))
     
     handle_errors(context, errors)
+
+def get_duplicate_values(values):
+    seen = set()
+    duplicates = [x for x in values if x in seen or seen.add(x)]
+    return duplicates
+
+def ifcopenshell_instance_type_to_string(v):
+    """
+        Converts ifcopenshell instance type to strings, if applicable
+    """
+    return do_try(lambda: v[0].is_a(), v)
```

Comment:
For documentation purposes can you explain why we need to do `v[0]` here? I don't understand.

Reply:
For example, the values for all instances of IfcGeometricRepresentationContext within a model could be (for example)
`[(#24=IfcMapConversion(#11,#23,316131.64,5690966.11,$,1.,0.,$),), (#25=IfcRigidOperation(#22,#23,$,$,$),), (#27=IfcMapConversion(#26,#23,316131.64,5690966.11,$,1.,0.,$),)]`

Reply:
It's now more general and readable, I think https://github.com/buildingSMART/ifc-gherkin-rules/commit/ca6d9d38e552cb690f6ccf3f32eb413ec577767b